boost-gradle/src/main/groovy/io/openliberty/boost/gradle/tasks/BoostPackageTask.groovy

```diff
@@ -167,6 +191,48 @@ public class BoostPackageTask extends AbstractBoostTask {
         return classifier
     }
 
+    protected List<BoosterPackConfigurator> getBoosterConfigsFromDependencies() {
+        List<String> dependencyList = new ArrayList<String>()
+
+        def componentIds = project.configurations.compile.incoming.resolutionResult.allDependencies.collect { it.selected.id }
+        def result = project.dependencies.createArtifactResolutionQuery()
+            .forComponents(componentIds)
+            .withArtifacts(MavenModule, MavenPomArtifact)
+            .execute()
+
+        for (component in result.resolvedComponents) {
+            logger.info("Adding ${component.id} to dependency list.")
+            component.getArtifacts(MavenPomArtifact).each { dependencyList.add(component.id.toString()) }
+        }
+
+        return boosterParent.mapDependenciesToFeatureList(dependencyList)
+    }
+
+    protected void generateServerXMLJ2EE(List<BoosterPackConfigurator> boosterConfigurators) throws GradleException {
+        try {
+            LibertyServerConfigGenerator serverConfig = new LibertyServerConfigGenerator(libertyServerPath)
+
+            // Add any other Liberty features needed depending on the boost
+            // boosters defined
+            List<String> boosterFeatureNames = getBoosterFeatureNames(boosterConfigurators)
+            serverConfig.addFeatures(boosterFeatureNames)
+            if (project.plugins.hasPlugin('war')) {
+                // write out config on behalf of a web app
+                serverConfig.addConfigForApp(project.war.baseName, project.war.version)
+            } else { // only support war packaging type currently
+                throw new GradleException(
+                        "Unsupported Gradle packaging type - Liberty Boost currently supports WAR packaging type only.");
+            }
+            serverConfig.addConfigForFeatures(boosterConfigurators)
+
+            // Write server.xml to Liberty server config directory
+            serverConfig.writeToServer()
+
+        } catch (TransformerException | IOException | ParserConfigurationException e) {
+            throw new GradleException("Unable to generate server configuration for the Liberty server.", e);
+        }
+    }
+
```

Comment:
This code is pretty key... since it's almost identical w/ Maven side except for exc handling, maybe it should be factored into common as well.

Reply:
Ya, there are some other changes we're going to make as part of moving the Booster stuff to common. We can refactor this part over with the other changes.